src/Hintman/HLint.hs

```diff
@@ -158,6 +166,71 @@ createCommentText content Idea{..} = do
         use ValidInstance{} = False -- Not important enough to tell an end user
         use _               = True
 
+{- | Line numbers in the file (modified by PR) and in the 'FileDelta' are
+are different as they have different meaning.
+
+HLint is working with the whole file and gives us the information about
+corresponding line in the file, while @git-diff@ only could show the modified
+chunks of lines. To leave a comment we can only apply them to the modified
+lines, so we need to know there number in the @git-diff@.
+
+This function is making this translation for you. Given a 'FileDelta' and the
+'Idea's last comment line number it returns the corresponding number of
+'FileDelta' which matches the last line of the HLint comment.
+
+If we have this modification in the file:
+
+@
+myFileDelta = FileDelta
+    { fileDeltaStatus = Modified
+    , fileDeltaSourceFile = "file.example"
+    , fileDeltaDestFile = "file.example"
+    , fileDeltaContent = Hunks
+        [ Hunk
+            { hunkSourceRange = Range
+                { rangeStartingLineNumber = 3
+                , rangeNumberOfLines = 2
+                }
+            , hunkDestRange = Range
+                { rangeStartingLineNumber = 3
+                , rangeNumberOfLines = 2
+                }
+            , hunkLines =
+                [ Line Context "Line number 3"
+                , Line Context "Line number 4"
+                ]
+@
+
+then it could only add HLint comment to lines 3 and 4 but not any other.
+
+>>> getTargetCommentPosition myFileDelta 3
+Just 0
+>>> getTargetCommentPosition myFileDelta 4
+Just 1
+>>> getTargetCommentPosition myFileDelta 5
+Nothing
+-}
+getTargetCommentPosition :: FileDelta -> Int -> Maybe Int
+getTargetCommentPosition FileDelta{..} commentPos = case fileDeltaContent of
+    Binary   -> Nothing
+    Hunks hs -> goHunks hs
+
+  where
+    -- Find the position number i the hunk.
+    -- Stops on the first satisfying.
+    goHunks :: [Hunk] -> Maybe Int
+    goHunks [] = Nothing
+    goHunks (Hunk{..}:hs) = whenNothing (inRange hunkDestRange) (goHunks hs)
```

Comment:
I also propose to return `Just` only on `Added` lines. We're not interested in `Removed` lines and I guess we don't want to give suggestions on `Context` because developers haven't touched them:

* http://hackage.haskell.org/package/diff-parse-0.2.1/docs/Text-Diff-Parse-Types.html#t:Annotation

What do you think?

Reply:
Okay, will do that check!